Extend support range for PHP

juicylime opened the pull request:
Signed-off-by: Aman Ahlwat <amanahlwat@gmail.com>

|    Q          |   A
|-------------- | ------

| New Feature   | yes


### Description

Extending the support range for PHP to support ~8.0

Fixes #24

Commit:
Extend support range for PHP

Signed-off-by: Aman Ahlwat <aman.ahlwat@grouphealth.ca>

Ocramius commented on composer.json:
```diff
@@ -29,7 +29,7 @@
     },
     "require": {
-        "php": "^7.3 || ~8.0.0",
+        "php": "^7.3 || ~8.0",
```
Should be `~7.3 || ~8.0.0 || ~8.1.0`

`8.2.0` onwards are still the wild west :-) 

A `composer update nothing` is also needed to update the `composer.lock` signature.

juicylime replied:
Ah yea that would make sense

Commit:
Only extend support range up to PHP 8.1

Ocramius commented:
Hmm, I think this error is because CI didn't run in a while, and `composer/composer:2.3.x` broke BC here.

`config.allow-plugins` is missing here: a local `composer install` will highlight that.

Commit:
Add config.allow-plugins to composer.json

Ocramius commented:
:thinking: ok, now that's where the mess starts: PHP 8.0 and 8.1 don't cleanly install at all? 

Nothing to do with your work, heh.

I see that `laminas/laminas-hydrator` support could be expanded to `^4` perhaps: `^3` is really really old. That would get rid of some of these composer issues: also here, I recommend a conservative `composer update nothing`

Commit:
Update some dependencies

juicylime commented:
There also looked to be a conflict with the laminas coding standard library so I went ahead and updated that too. Hopefully that wont pose any issues?

juicylime commented:
Ah I found another PR that addresses some of the issues going on here https://github.com/laminas-api-tools/api-tools-doctrine-querybuilder/pull/23. The PR is still open so maybe there's some outstanding issues with the dependencies?

juicylime commented:
Looks like laminas-api-tools/api-tools-doctrine only supports until PHP 8.0 so something similar would need to be done there to extend it to 8.1 (If it doesn't break other packages in the process) and the PHP 8.1 tests would pass here once that package has been updated to support 8.1 as well. 

I'm not so sure about the PHP 8.0 tests since there are issues with the phpro/zf-doctrine-hydration-module dependency and that only supports up until PHP ^7.0
